src/openpgp/packet/keymaterial.iced

```diff
@@ -479,7 +515,7 @@ class KeyMaterial extends Packet
         expire_at = generated + expire_in
         if not winner? or (winner.expire_at? and (winner.expire_at < expire_at))
           winner = { expire_at, generated, expire_in }
-      else if (expire_in? and expire_in is 0) or (not expire_in? and not @is_primary())
```

Comment:
I thought the timeline diagram was about newly-added expiration dates on one uid overriding other non-expiring uids. But that doesn't seem to be what this function was doing before. Here it looks like it will always let the non-expiring date win (if we remove the `is_primary` check).

Was the worry that doing this would be too permissive?